File: bridge/src/bridge_social.cpp
#if defined(DM_PLATFORM_HTML5)
#include "bridge_social.h"
#include "bridge.h"

// Share
int bridge::social::isShareSupported(lua_State* L) {
    return getBoolean(L, js_bridge_social_isShareSupported);
}

int bridge::social::share(lua_State* L) {
    return makeCallbackWithJson(L, js_bridge_social_share, false);
}

// Join Community
int bridge::social::isJoinCommunitySupported(lua_State* L) {
    return getBoolean(L, js_bridge_social_isJoinCommunitySupported);
}

int bridge::social::joinCommunity(lua_State* L) {
    return makeCallbackWithJson(L, js_bridge_social_joinCommunity, false);
}

// Invite Friends
int bridge::social::isInviteFriendsSupported(lua_State* L) {
    return getBoolean(L, js_bridge_social_isInviteFriendsSupported);
}

int bridge::social::inviteFriends(lua_State* L) {
    return makeCallbackWithJson(L, js_bridge_social_inviteFriends, false);
}

// Create Post
int bridge::social::isCreatePostSupported(lua_State* L) {
    return getBoolean(L, js_bridge_social_isCreatePostSupported);
}

int bridge::social::createPost(lua_State* L) {
    return makeCallbackWithJson(L, js_bridge_social_createPost, false);
}

// Add to Favorites
int bridge::social::isAddToFavoritesSupported(lua_State* L) {
    return getBoolean(L, js_bridge_social_isAddToFavoritesSupported);
}

int bridge::social::addToFavorites(lua_State* L) {
    return makeCallback(L, js_bridge_social_addToFavorites, false);
}

// Add to HomeScreen
int bridge::social::isAddToHomeScreenSupported(lua_State* L) {
    return getBoolean(L, js_bridge_social_isAddToHomeScreenSupported);
}

int bridge::social::addToHomeScreen(lua_State* L) {
    return makeCallback(L, js_bridge_social_addToHomeScreen, false);
}

// Rate Game
int bridge::social::isRateSupported(lua_State* L) {
    return getBoolean(L, js_bridge_social_isRateSupported);
}

int bridge::social::rate(lua_State* L) {
    return makeCallback(L, js_bridge_social_rate, false);
}

// External Links
int bridge::social::isExternalLinksAllowed(lua_State* L) {
    return getBoolean(L, js_bridge_social_isExternalLinksAllowed);
}

#endif

File: bridge/bridge_social.lua
local social = {}

-- Share
function social.is_share_supported()
	return false
end

function social.share(options, on_success, on_failure)
	if on_failure then
		on_failure()
	end
end

-- Join Community
function social.is_join_community_supported()
	return false
end

function social.join_community(options, on_success, on_failure)
	if on_failure then
		on_failure()
	end
end

-- Invite Friends
function social.is_invite_friends_supported()
	return false
end

function social.invite_friends(options, on_success, on_failure)
	if on_failure then
		on_failure()
	end
end

-- Create Post
function social.is_create_post_supported()
	return false
end

function social.create_post(options, payload, on_success, on_failure)
	if type(payload) == "function" then
		on_failure = on_success
		on_success = payload
	end
	if on_failure then
		on_failure()
	end
end

-- Add to Favorites
function social.is_add_to_favorites_supported()
	return false
end

function social.add_to_favorites(on_success, on_failure)
	if on_failure then
		on_failure()
	end
end

-- Add to Home Screen
function social.is_add_to_home_screen_supported()
	return false
end

function social.add_to_home_screen(on_success, on_failure)
	if on_failure then
		on_failure()
	end
end

-- Rate Game
function social.is_rate_supported()
	return false
end

function social.rate(on_success, on_failure)
	if on_failure then
		on_failure()
	end
end

-- Post Reward
function social.is_post_reward_supported()
	return false
end

function social.get_post_reward(on_success, on_failure)
	if on_failure then
		on_failure()
	end
end

return social